artists/views.py

```diff
@@ -2,44 +2,69 @@
 from django.shortcuts import render
 
 from albums.models import Album
-from albums.views import show
 
 from django.core.exceptions import ValidationError
 
+from django.views import View
+
 from .models import Artist
 
+from django.http import HttpResponse
+
+class Index(View):
+    def getData(self ,*args, **kwargs):
+        artists = []
+        for artist in Artist.objects.all():
+            artists.append({'artist':artist,'albums':Album.objects.filter(Q(artist=artist))})  
+        
+        # for item in albums[44]:
+        #     print(item.name)
+
+        data = {
+            'artists' : artists,
+            'msg' : kwargs['data']
+            # 'albums': albums
+            # 'albums' : Artist.objects.filter(Q(album__is_approved=True)).annotate(count = Count('album')).order_by('-count')
+        }
+        return data 
+    def post (self,request,*args,**kwargs):
+        return render(request,"artists/index.html",self.getData(data = kwargs['data']['msg']))
+         
+    def get(self,request,*args,**kwargs):
+        return render(request,"artists/index.html",self.getData(data = ''))
+
+class Create(View):
```

Comment:
```python
class Create(LoginRequiredMixin, CreateView):
    model = Artist
    fields = ['stage_name', 'social_link']
    template_name = 'artists/create.html'
```

Reply:
what is the point of using this ? , i used different way but , what's better ?